### PR TITLE
2460 Refactor module to integrate with ortograf service

### PR DESCRIPTION
**Pull request by googletorp**
Split module into two, core module for integration and
a UI module for the actual autocomplete functionality.

Note for Netpunkt - this needs https://git.dbc.dk/netpunkt/netpunkt/pull-request/750/_/feature/2460_bibdk_ortograf_refactor as well
This should be able to be used as is on bib.dk, but will require some config update. Also bibdk_frontend adds a JS file now included in bibdk_autocomplete_ui so not needed there anymore.

For Netpunkt important fact is that ortograf integration does not currently support filtering on material type which is needed for some of the autocompletes on netpunkt. I don't know if this is a limitation in the service, or if this has not been needed on bibdk. As there was no docs on ortograf service, I could not solve this issue.

This has been tested on netpunkt but not on bibdk. Other than some cleanup etc, should be a mirror of what bibdk currently has, so this is a good opportunity to have both systems use same module.

### CODE REVIEW

File: bibdk_autocomplete_ui/bibdk_autocomplete_ui.admin.inc [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+    '#type' => 'fieldset',
+    '#title' => t('Bibliotek.dk autocomplete fields', array(), array('context' => 'bibdk_autocomplete_ui')),
+    '#tree' => TRUE,
+  );
+
+  $base_vars['type'] = 'all';
+
+  foreach ($result as $value) {
+
+    if ($value->value_type != 'textfield') {
```
Comment (dbclkh) [inline bibdk_autocomplete_ui/bibdk_autocomplete_ui.admin.inc:77]: Hvad gør denne foreach løkke?

File: bibdk_autocomplete_ui/js/bibdk_autocomplete_ui.js [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+        $(this.popup).css({ visibility: 'hidden' });
+        this.hidePopup();
+      }
+    }
+  };
+
+  /**
+   * overrides for drupal autocomplete w/o ajax
+   */
+  if (Drupal.jsAC){
```
Comment (dbclkh) [inline bibdk_autocomplete_ui/js/bibdk_autocomplete_ui.js:191]: Mellemrum

File: bibdk_autocomplete_ui/js/bibdk_autocomplete_ui.js [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+   */
+  BibdkAutocompleteBehavior.submitValues = function (elemId, elemValue) {
+    this.fields[elemId].submitValue = elemValue;
+  };
+
+  /**
+   * Split autocomplete url and find the filter values
+   *
+   * @param url
+   * @returns {*[]}
```
Comment (dbclkh) [inline bibdk_autocomplete_ui/js/bibdk_autocomplete_ui.js:122]: a'hva returnere den?

File: bibdk_autocomplete_ui/bibdk_autocomplete_ui.admin.inc [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+      '#tree' => TRUE,
+      '#collapsible' => TRUE,
+    );
+
+    if ((empty($default_value['autocomplete']))) {
+      $form['bibdk_autocomplete_ui_var'][$value->p_uuid]['vars'][$value->v_uuid]['#collapsed'] = TRUE;
+    }
+
+    $form['bibdk_autocomplete_ui_var'][$value->p_uuid]['page_path'] = array(
+      '#type' => 'hidden',
```
Comment (dbclkh) [inline bibdk_autocomplete_ui/bibdk_autocomplete_ui.admin.inc:95]: Bruger du denne page_path til JS eller lign. siden det er en hidden type og ikke bare en value?

**Review by dbclkh — CHANGES_REQUESTED** [review]
Jeg afviser denne Code Review pga. den manglende TingClient, ikke resten af din kode. Lad endelig denne feature branch ligge - vi tager fat i den på et senere tidspunkt.

**Comment by dbclkh** [comment]
Denne pull request må ikke accepteres da den er sat til at lande i MASTER.

File: bibdk_autocomplete_ui/bibdk_autocomplete_ui.admin.inc (hunk 1 above)
Comment (googletorp) [reply bibdk_autocomplete_ui/bibdk_autocomplete_ui.admin.inc:77]: JS fil er samme som: https://github.com/DBCDK/bibdk/blob/master/modules/bibdk_autocomplete/js/bibdk_autocomplete.ortograf.js

File: bibdk_autocomplete_ui/bibdk_autocomplete_ui.admin.inc (hunk 4 above)
Comment (googletorp) [reply bibdk_autocomplete_ui/bibdk_autocomplete_ui.admin.inc:95]: Jeg har taget dette fra: https://github.com/DBCDK/bibdk/blob/master/modules/bibdk_autocomplete/includes/bibdk_autocomplete.admin.ortograf.inc

**Comment by googletorp** [comment]
@dbclkh Jeg antog at det var et bevidst valg ikke at bruge ting client i https://github.com/DBCDK/bibdk/blob/master/modules/bibdk_autocomplete/bibdk_autocomplete.module hvor selve integrationskoden er taget fra

**Pull request closed without merging**